### PR TITLE
Fix a bug in maskdef_offset

### PR DESCRIPTION
**Pull request by debora-pe**
I found a bug in how the `maskdef_offset` was assigned to each detector. Now the bug is fixed.

I also changed the default behavior for how DEIMOS computes the `maskdef_offset`. It now uses by default the stars in the alignment boxes.

### CODE REVIEW

**Comment by debora-pe** [comment]
Tests pass (run only the ones that touch this part of the code)

Dev Suite:
```
./pypeit_test develop -i keck_deimos -t 2
Running tests on the following instruments:
    keck_deimos

Reducing data from keck_deimos for the following setups:
    600ZD_M_6500
    600ZD_tilted
    1200G_M_7750
    830G_LVM_8400
    830G_M_8100_26
    830G_M_8500
    830G_L_8100
    1200B_M_5200
    1200G_M_5500
    900ZD_M_6000
    1200B_LVM_5200
    900ZD_LVM_5500
    830G_M_9000_dither

Running tests in 2 parallel processes
 0 passed/ 0 failed/ 0 skipped STARTED keck_deimos/600ZD_M_6500 pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_8500 pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_8500 pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_8500 pypeit_multislit_flexure
 2 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_8500 pypeit_multislit_flexure
 2 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200B_M_5200 pypeit (without masters)
 3 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/600ZD_M_6500 pypeit (without masters)
 3 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200G_M_7750 pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200G_M_7750 pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped STARTED keck_deimos/900ZD_M_6000 pypeit (without masters)
 5 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200B_M_5200 pypeit (without masters)
 5 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_9000_dither pypeit (without masters)
 6 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_9000_dither pypeit (without masters)
 6 passed/ 0 failed/ 0 skipped STARTED keck_deimos/600ZD_tilted pypeit (without masters)
 7 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/900ZD_M_6000 pypeit (without masters)
 7 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_L_8100 pypeit (without masters)
 8 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_L_8100 pypeit (without masters)
 8 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200G_M_5500 pypeit (without masters)
 9 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/600ZD_tilted pypeit (without masters)
 9 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_LVM_8400 pypeit (without masters)
10 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200G_M_5500 pypeit (without masters)
10 passed/ 0 failed/ 0 skipped STARTED keck_deimos/900ZD_LVM_5500 pypeit (without masters)
11 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_LVM_8400 pypeit (without masters)
11 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_8100_26 pypeit (without masters)
12 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/900ZD_LVM_5500 pypeit (without masters)
12 passed/ 0 failed/ 0 skipped STARTED keck_deimos/900ZD_LVM_5500 pypeit_sensfunc
13 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/900ZD_LVM_5500 pypeit_sensfunc
13 passed/ 0 failed/ 0 skipped STARTED keck_deimos/900ZD_LVM_5500 pypeit_flux
14 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/900ZD_LVM_5500 pypeit_flux
14 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200B_LVM_5200 pypeit (without masters)
15 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_8100_26 pypeit (without masters)
16 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200B_LVM_5200 pypeit (without masters)

--- PYPEIT DEVELOPMENT SUITE PASSED 16/16 TESTS (Masters ignored) ---
Testing Started at 2021-12-06T15:56:14.377367
Testing Completed at 2021-12-06T19:02:31.050911
```

And 

```
./pypeit_test develop -i keck_mosfire -t 2
Running tests on the following instruments:
    keck_mosfire

Reducing data from keck_mosfire for the following setups:
    Y_long
    J_multi
    K_long
    Y_multi
    long2pos1_H
    longslit_3x0.7_H
    mask1_K_with_continuum
    mask1_J_with_continuum

Running tests in 2 parallel processes
 0 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/Y_multi pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/J_multi pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/J_multi pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/mask1_J_with_continuum pypeit (without masters)
 2 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/mask1_J_with_continuum pypeit (without masters)
 2 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/mask1_K_with_continuum pypeit (without masters)
 3 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/Y_multi pypeit (without masters)
 3 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/Y_long pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/mask1_K_with_continuum pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/K_long pypeit (without masters)
 5 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/Y_long pypeit (without masters)
 5 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/Y_long pypeit_sensfunc
 6 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/Y_long pypeit_sensfunc
 6 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/Y_long pypeit_ql
 7 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/Y_long pypeit_ql
 7 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/long2pos1_H pypeit (without masters)
 8 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/K_long pypeit (without masters)
 8 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/longslit_3x0.7_H pypeit (without masters)
 9 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/longslit_3x0.7_H pypeit (without masters)
10 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/long2pos1_H pypeit (without masters)

--- PYPEIT DEVELOPMENT SUITE PASSED 10/10 TESTS (Masters ignored) ---
```

Unit tests:
```
pytest test_slitmask.py -W ignore
================================================================================================= test session starts =================================================================================================
platform darwin -- Python 3.8.5, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/dpelliccia/PypeIt/PypeIt, configfile: setup.cfg
collected 3 items                                                                                                                                                                                                     

test_slitmask.py ...                                                                                                                                                                                            [100%]

============================================================================================ 3 passed in 187.40s (0:03:07) ============================================================================================

```

File: pypeit/slittrace.py [hunk 1]
```diff
@@ -1476,7 +1477,7 @@ def average_maskdef_offset(calib_slits, platescale, list_detectors):
     indx_b = np.where(np.in1d(calib_dets, spectrograph_dets[0]))[0]
     # if this spectrograph is not split into blue and red detectors
     # or if it is but there are no available offsets in the blue
-    if not blue_and_red or slitmask_offsets[indx_b].size == 0:
+    if not blue_and_red or indx_b.size == 0:
```
Comment (profxj) [inline pypeit/slittrace.py:1480]: Am wondering how this will go (or go away) once 
we start mosaicing..

**Review by profxj — APPROVED** [review]
Good by me.  Thx

**Review by kbwestfall — APPROVED** [review]
Looks good to me.  Please merge the `release` branch into this one before merging.

File: pypeit/slittrace.py (hunk 1 above)
Comment (debora-pe) [reply pypeit/slittrace.py:1480]: Good point! We may need to make some changes to this part of the code. I'll keep in mind.

**Comment by debora-pe** [comment]
Thanks @kbwestfall and @profxj !
I'm merging.

**Pull request merged**